netlify/netlify-plugin-gatsby
# feat: support for Gatsby v5

### Summary

Include .nvmrc files in both demos to specify minimum node versions in order to make it easy to work with now that there are different minimum node versions for the 2 demos.

I opted to create a 2nd demo site as we will have to continue to support v4 for the time being (will be until end of 2023 as that is how long Gatsby will maintain it, though will only have maintenance changes done to it).

I've also updated the minimum version of react to 18 in e2e tests that have `gatsby@next` being installed. We may want to consider having more e2e tests to ensure v4 functionality is maintained, but at the moment e2e tests take a while to run and I wanted to have that discussion with the broader team.

Dependant PR: https://github.com/netlify/gatsby-plugin-netlify/pull/203

Migration guide from Gatsby - https://www.gatsbyjs.com/docs/reference/release-notes/migrating-from-v4-to-v5/

### Test plan

1. Visit the Deploy Preview - https://ep-gatsby-v5--ep-gatsby-v5.netlify.app
* Confirm that the SSR and DSG pages continue to work as expected

2. Visit the v4 deploy preview - https://deploy-preview-547--netlify-plugin-gatsby-demo.netlify.app/
* Confirm the links work as expected
* Visit the `api/users/123` to confirm that the page doesn't crash (this confirms the updated import logic [here](https://github.com/netlify/netlify-plugin-gatsby/pull/547/files#diff-5961929d584c53e46053694aeb9683e9ca1b4ca27dcf87a2d513690428d63ac9R23) works as expected

### Relevant links (GitHub issues, Notion docs, etc.) or a picture of cute animal
Fixes https://github.com/netlify/pod-ecosystem-frameworks/issues/269 , https://github.com/netlify/gatsby-plugin-netlify/issues/195

### Standard checks:
